lib/Controller/MessagesController.php

```diff
@@ -243,6 +243,7 @@ public function getBody(int $id): JSONResponse {
 		$json['isSenderTrusted'] = $this->isSenderTrusted($message);
 
 		$smimeData = new SmimeData();
+		$smimeData->setIsEncrypted($message->isEncrypted());
 		try {
 			$parsedMessage = Horde_Mime_Part::parseMessage($rawMessage, ['no_body' => true]);
```

Comment:
I fixed it for opaque signed messages `smime-type="signed-data"`. The code for regular signed messages should be moved to the `ImapMessageFetcher` too. I already added TODOs to the code.

IMO, we can move this to a follow up because the encrypted lock will be shown and takes precedence over he signed lock.